### PR TITLE
feat(signature): add POST/PATCH methods for the signature

### PR DESCRIPTION
**Pull request by ccoeurderoy**
## Description

Signatures API

## Motivation and Context

- Add a static `create` method to be able to call `POST /folders/:id/signatures`
- Add an `update` method to be able to call `PATCH /folders/:id//signatures/:signatureId`

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: src/core/Folder.ts [hunk 1]
```diff
@@ -35,7 +34,7 @@ export class Folder implements IFolder {
 
   public legalDocuments: LegalDocument[];
 
-  public signatures: Signature[];
+  public signatures: ISignature[];
```
Comment (amarlankri) [inline src/core/Folder.ts]: Why did you change that? With this change, is it still possible to call `Signature#update`?

File: test/signature.test.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        legalDocumentIds: ['random_legal_id'],
+        holder: Holder.APPLICANT,
+      });
+      expect(signatureAPI.isDone()).toBeTruthy();
+      expect(signature).toBeInstanceOf(Signature);
+      expect(signature.id).toEqual('5f3663fc3ec5884afc8ed04f');
+      expect(signature.legalDocumentIds).toEqual(['random_legal_id']);
+      expect(signature.partnerId).toEqual('SIGN_ID');
+      expect(signature.holder).toEqual('APPLICANT');
+      expect(signature.createdAt).toEqual(new Date('2020-08-14T10:14:20.224+0000'));
```
Comment (amarlankri) [inline test/signature.test.ts:61]: What is the interest of these checks?

For me, these statements test fake server rather than our API. Am I wrong?

File: src/core/Signature.ts [hunk 3]
```diff
@@ -74,4 +81,67 @@ export class Signature implements ISignature {
...
+
+  /**
+   * Get a signature instance from a folder by id
+   * @param folderId Unique folder identifier
+   * @param signatureId Unique signature identifier
+   */
+  public static async getById(
+    requestBuilder: RequestBuilder,
+    folderId: string,
+    signatureId: string,
```
Comment (amarlankri) [inline src/core/Signature.ts]: If forgot a remark.

Maybe it is better to pass these two arguments as object. It is what I did in my side because I think it is less error prone than passing two arguments of the same type (if the order is false, the request is wrong).

What do you think?

File: src/core/Folder.ts (hunk 1 above)
Comment (ccoeurderoy) [reply src/core/Folder.ts]: You're right, I shouldn't

File: src/core/Signature.ts (hunk 3 above)
Comment (ccoeurderoy) [reply src/core/Signature.ts]: I think you"re right, I'm correcting this

**Review by amarlankri — APPROVED** [review]
